tests/js/spec/redux/reducers/shared.spec.js

```diff
@@ -0,0 +1,234 @@
+import {Set} from 'immutable';
+import {resource} from 'app/redux/reducers/shared';
+import {makeResourceActions} from 'app/redux/actions/shared';
+
+const RESOURCE_NAME = 'SOME_RESOURCE';
+
+// Tests the shared functionality of resource reducers
+const initialState = {
+  ...resource.initialState,
+};
+
+const reducer = resource.createReducer(RESOURCE_NAME, initialState);
+const actions = makeResourceActions(RESOURCE_NAME, '/api/0/some-resource/');
+
+function createGetListSuccessState() {
+  const originalState = {...initialState};
+  const requestedState = reducer(
+    originalState,
+    actions.getListRequest('search', 'groupby', 'cursor')
+  );
+  return reducer(
+    requestedState,
+    actions.getListSuccess(
+      [
+        {id: '1', name: 'entry1'},
+        {id: '2', name: 'entry2'},
+      ],
+      'link-to-more-results'
+    )
+  );
+}
+
+describe('shared resource reducer', () => {
+  it('has expected state after requesting a list', () => {
+    const originalState = {...initialState};
+    const requestedState = reducer(
+      originalState,
+      actions.getListRequest('search', 'groupby', 'cursor')
+    );
+    expect(requestedState).toEqual({
+      loadingDetails: false,
+      detailsId: null,
+      loading: true,
+      errorMessage: null,
+      byIds: {},
+      listViewState: {
+        allVisibleSelected: false,
+        groupBy: 'groupby',
+        search: 'search',
+        visibleIds: [],
+        selectedIds: new Set(),
+        pagination: {pageLinks: null, cursor: 'cursor'},
+      },
+      creating: false,
+    });
+  });
+
+  it('has expected state when getting a successful list response', () => {
+    const successState = createGetListSuccessState();
+    expect(successState).toEqual({
+      loadingDetails: false,
+      detailsId: null,
+      loading: false,
+      errorMessage: null,
+      byIds: {'1': {id: '1', name: 'entry1'}, '2': {id: '2', name: 'entry2'}},
```

Comment:
When coming to expandable lists, there is a lot of metadata containing UI state, such as isFetched, cachedIds, etc. The api entries are therefore embedded into "entity" attribute, to be distinguished from ui metadata. This make the shared reducers not compatible with expandable list reducers. I thought you should know that.

Reply:
Right, thanks. I think it will be some work to merge the two. I've created a ticket on it.